auth/src/main/java/software/amazon/awssdk/auth/credentials/HttpCredentialsProvider.java

```diff
@@ -116,12 +129,15 @@ private Instant getPrefetchTime(Instant expiration) {
 
     @Override
     public AwsCredentials getCredentials() {
-        return credentialsCache.get();
+        if (isLocalCredentialLoadingDisabled()) {
+            return null;
```

Comment:
We currently throw exceptions instead of returning null from credential providers, specifying why the credentials couldn't be loaded.

Reply:
Good catch. Updated